### DISCUSSION TITLE
Current_NotificationRecieved function not running when app is closed

### DISCUSSION THREAD

**Discussion by David2491** [thudugala/Plugin.LocalNotification#373] [Q&A] [answered]
Hello, I am using LocalNotificationCenter.Current.NotificationRecieved to perform some code once a notification goes off on the user's device. The only issue is it does not run when the app is closed. If the app is open when the notification triggers, the NotificationRecieved function runs, if not then it will not run and the code needed doesn't execute, does anyone know what I should put in my code so that the function will always run even when the app is closed. Thanks

**Comment by thudugala** [comment, marked as answer]
@david20711

If the app is inactive, the app is not listening to NotificationReceived event. 

The current design does not support what you need to do.

**Comment by ckoziel** [comment]
Can we add the Plugin.LocalNotification source to a project, and then embed the DLL into our project? Therefore, we can make changes to the NotificationReceived event?

**Reply by thudugala** [reply]
@ckoziel I do not think that will work. the app is not listening to NotificationReceived event
